From 62fd02a727ae6fdfaac3fb2cd18cf36412153a0e Mon Sep 17 00:00:00 2001
From: Felix Yan <felixonmars@archlinux.org>
Date: Fri, 18 Sep 2015 14:07:51 +0800
Subject: [PATCH] add support for xcffib as an alternative to xpyb

xcffib is well maintained and packaged for most Linux distros, and it is
meant to be mostly a drop-in replacement to xpyb. It would be great if
xcffib support could be added so we don't need to package the legacy
xpyb.

I have also filed a pull request for xpybutil to add xcffib support at
https://github.com/BurntSushi/xpybutil/pull/8

Change-Id: I9491b5383d6df2ee214e804d93903f91484be076
---
 src/views/window.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/views/window.py b/src/views/window.py
index 18600d0..181d5c0 100644
--- a/src/views/window.py
+++ b/src/views/window.py
@@ -24,7 +24,10 @@
 import json
 from random import randint
 
-import xcb
+try:
+    from xcb.xproto import PropMode
+except ImportError:
+    from xcffib.xproto import PropMode
 from xpybutil.icccm import State
 from xpybutil.ewmh import (c, atom, request_wm_state_checked,
     request_active_window_checked, revent_checked )
@@ -118,7 +121,7 @@ def play(self, pathList):
     def setDeepinWindowShadowHint(self, width):
         width = str(width)
         window = self.winId().__int__()
-        return c.core.ChangeProperty(xcb.xproto.PropMode.Replace, window,
+        return c.core.ChangeProperty(PropMode.Replace, window,
                                      atom('DEEPIN_WINDOW_SHADOW'),
                                      atom('STRING'), 8, len(width), width)
 
